kernel/constr.mli

```diff
@@ -138,14 +138,30 @@ val mkRef : GlobRef.t Univ.puniverses -> constr
 
 (** Constructs a destructor of inductive type.
     
-    [mkCase ci p c ac] stand for match [c] as [x] in [I args] return [p] with [ac] 
+    [mkCase ci params p c ac] stand for match [c] as [x] in [I args] return [p] with [ac]
     presented as describe in [ci].
 
-    [p] stucture is [fun args x -> "return clause"]
+    [params] are the parameter arguments of [c]
+
+    [p] stucture is [args x |- "return clause"]
 
     [ac]{^ ith} element is ith constructor case presented as 
-    {e lambda construct_args (without params). case_term } *)
-val mkCase : case_info * constr * constr * constr array -> constr
+    {e construct_args |- case_term } *)
+
+type 'constr pcase_branch = Name.t Context.binder_annot array * 'constr
+(** Names of the indices + name of self *)
+
+type 'types pcase_return = Name.t Context.binder_annot array * 'types
+(** Names of the branches *)
```

Comment:
Is it OK to copy the relevances here? Maybe they should be gotten from the inductive_body like with the types.

Reply:
No, otherwise you'd need to fetch them during reduction, always, and that would incur an important overhead for no good reason.

Reply:
Conversion not reduction right? I guess the current lookup is only on polymorphic inductives (to check variance) which is still rare.

Reply:
(I didn't mean specifically the _return one, I just had to pick one in the github interface)

Reply:
You still need to push the relevance of binders in the environment during conversion though, so you need to have them somewhere.